### ISSUE TITLE
Enum member type hints in third party code should cause no issues in application code

### ISSUE THREAD

**Issue by RomanValov** [microsoft/pyright#11452]
**Describe the bug**

I'm using third-party library which erroneously uses type hints for `StrEnum` members.
It's reasonable `pyright` complains about these unnecessary type hints for enum members when type checking library code (due to #8060).
However `pyright` also complains against my application code which uses the library in the sane way.

**Code or Screenshots**

./third_party.py

```python
from dataclasses import dataclass
from enum import StrEnum


class Option(StrEnum):
    YES: str = 'YES'
    NO: str = 'NO'


@dataclass
class Config:
    option: Option
```

./application.py

```python
from third_party import Option
from third_party import Config

if __name__ == '__main__':
    print(Config(Option.YES))
```

**Observed behavior**

```
$ pyright ./application.py
./application.py
  ./application.py:5:18 - error: Argument of type "str" cannot be assigned to parameter "option" of type "Option" in function "__init__"
    "str" is not assignable to "Option" (reportArgumentType)
1 errors, 0 warnings, 0 informations
```

**Expected behavior**

No issues for application.py

**VS Code extension or command-line**

pyright 1.1.409

**Comment by rchiodo** [comment]
I think this would be by design. You're using an invalid enum. I don't think Pyright would special case this such that it would ignore the declared type on the member.
